paper-api/src/main/java/org/bukkit/EntityEffect.java

```diff
@@ -1,122 +1,136 @@
      * @see #ENTITY_DEATH
      */
-    @Deprecated(since = "1.12.2")
+    @Deprecated(since = "1.12.2", forRemoval = true)
     DEATH(3, Entity.class),
     /**
-     * Spawns the egg breaking particles
+     * Spawns the egg breaking particles.
+     *
+     * @deprecated use {@link #PROJECTILE_CRACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     EGG_BREAK(3, Egg.class),
     /**
-     * Spawns the snowball breaking particles
+     * Spawns the snowball breaking particles.
+     *
+     * @deprecated use {@link #PROJECTILE_CRACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     SNOWBALL_BREAK(3, Snowball.class),
     /**
-     * Plays the entity death sound and animation
+     * Shows the crack particles when a projectile
+     * hits something.
+     */
+    PROJECTILE_CRACK(3, Egg.class, Snowball.class),
+    /**
+     * Plays the entity death sound and animation.
      * <p>
      * <b>This will cause client-glitches!</b>
      */
     ENTITY_DEATH(3, LivingEntity.class),
     /**
-     * Plays the fang attack animation
+     * Plays the evoker's fang attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     FANG_ATTACK(4, EvokerFangs.class),
     /**
-     * Plays the hoglin attack animation
+     * Plays the hoglin attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     HOGLIN_ATTACK(4, Hoglin.class),
     /**
-     * Plays the iron golem attack animation
+     * Plays the iron golem attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     IRON_GOLEN_ATTACK(4, IronGolem.class),
     /**
-     * Plays the ravager attack animation
+     * Plays the ravager attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     RAVAGER_ATTACK(4, Ravager.class),
     /**
-     * Plays the warden attack animation
+     * Plays the warden attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     WARDEN_ATTACK(4, Warden.class),
     /**
-     * Plays the zoglin attack animation
+     * Plays the zoglin attack animation.
+     *
+     * @deprecated use {@link #ENTITY_ATTACK}
      */
+    @Deprecated(since = "1.21.4", forRemoval = true)
     ZOGLIN_ATTACK(4, Zoglin.class),
+    /**
+     * Plays an attack animation for the respective entities.
+     */
+    ENTITY_ATTACK(4, EvokerFangs.class, Hoglin.class, IronGolem.class, Ravager.class, Warden.class, Zoglin.class, Creaking.class),
     // 5 - unused
     /**
      * The smoke when taming an entity fails.
+     *
      * @deprecated use {@link EntityEffect#TAMING_FAILED}
      */
-    @Deprecated(since = "1.21") // Paper
+    @Deprecated(since = "1.21")
```

Comment:
Whats the reasoning beyond marking other "renames" as forRemoval but this one isn't?